index.js

```diff
@@ -356,6 +356,19 @@
     ($.Undefined  = type0('Undefined'))
   ];
 
+  // Enum :: [Any] -> Type
+  $.Enum = function(constants) {
+    return {
+      type: 'sanctuary-def/Enum',
+      test: function(x) {
+        return $.Array(Any).test(constants) && constants.indexOf(x) !== -1;
```

Comment:
I don't think that last bit will work because `reprs` depends on `show(x)` and `x` is only available in `test`'s scope.